examples/cluster/plot_coin_segmentation.py

```diff
@@ -63,27 +66,39 @@
 graph.data = np.exp(-beta * graph.data / graph.data.std()) + eps
 
 # Apply spectral clustering (this step goes much faster if you have pyamg
-# installed)
-N_REGIONS = 25
+# installed and use eigen_solver = 'amg'). However, any valid solver can
+# be used (e.g., 'arpack', 'lobpcg', or 'amg').
+eigen_solver = 'arpack'
+
+# The number of regions
+n_regions = 26
 
 # %%
-# Visualize the resulting regions
+# Compute and visualize the resulting regions
 
-for assign_labels in ('kmeans', 'discretize'):
+# Any eigen_solver: 'arpack', 'lobpcg', 'amg' can be used. AMG is usually best
+# It often helps the spectral clustering to compute a few extra eigenvectors
+n_regions_plus = 3
```

Comment:
Note that this makes also make the `discretize` method work much better than on `main` from a qualitative point of view.